dowhy/gcm/_noise.py

```diff
@@ -20,3 +40,108 @@ def compute_noise_from_data(causal_model: InvertibleStructuralCausalModel, obser
                     observed_data[get_ordered_predecessors(causal_model.graph, node)].to_numpy()))
 
     return result
+
+
+def get_noise_dependent_function(causal_model: StructuralCausalModel,
```

Comment:
this file was already reviewed in #452 Good to merge that with that PR/master

Reply:
Rebased it accordingly. This should not be part of the PR anymore.